src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,37 @@
      * Андрей - 600
      * София - 900</p>
      */
+
+    private static final DateTimeFormatter FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        LocalDate firstDay = LocalDate.parse(dateFrom,FORMATTER);
+        LocalDate endDay = LocalDate.parse(dateTo, FORMATTER);
+        if (firstDay.isAfter(endDay)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuilder result = new StringBuilder();
+        result.append("Отчёт за период ").append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            int money = 0;
+            for (String salaryInfo : data) {
+                //int money =0;
```

Comment:
do we need this comment? let's remove it))

Reply:
done